### PR TITLE
This adds the support for repo search in github related tasks

### PR DESCRIPTION
**Pull request by romil07**

### CODE REVIEW

File: Tasks/DownloadGitHubReleaseV0/task.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "demands": [],
   "version": {
     "Major": 0,
-    "Minor": 148,
+    "Minor": 149,
```
Comment (gykuma) [inline Tasks/DownloadGitHubReleaseV0/task.json]: If these changes are to be checked-in in M148 we should only change the patch version

File: Tasks/DownloadGitHubReleaseV0/task.json [hunk 2]
```diff
@@ -94,6 +99,15 @@
...
         "releaseByTagName": "$(defaultVersionType)"
       }
+    },
+    {
+      "target": "userRepositorySearch",
+      "endpointId": "$(connection)",
+      "dataSourceName": "SearchRepositoriesByName",
+      "resultTemplate": "{ \"Value\" : \"{{full_name}}\", \"DisplayValue\" : \"{{full_name}}\" }",
+      "parameters": {
+        "searchText": "$(name)"
```
Comment (gykuma) [inline Tasks/DownloadGitHubReleaseV0/task.json]:
```suggestion
        "searchText": "$(version)"
```

File: Tasks/DownloadGitHubReleaseV0/task.loc.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   "demands": [],
   "version": {
     "Major": 0,
-    "Minor": 148,
+    "Minor": 156,
```
Comment (gykuma) [inline Tasks/DownloadGitHubReleaseV0/task.loc.json]: this should be same as in **task.json** (auto generated on build)

File: Tasks/Common/utility-common/package-lock.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
       "resolved": "https://registry.npmjs.org/argparse/-/argparse-1.0.10.tgz",
```
Comment (gykuma) [inline Tasks/Common/utility-common/package-lock.json]: changes in this file doesn't seem related, should we ignore it?

Comment (romil07) [reply Tasks/Common/utility-common/package-lock.json]: Yes. These won't be checked in.

File: Tasks/DownloadGitHubReleaseV0/task.loc.json (hunk 3 above)
Comment (romil07) [reply Tasks/DownloadGitHubReleaseV0/task.loc.json]: I did not rebuild the task before the final check in. Hence, this change. It would be 149 otherwise.

File: Tasks/DownloadGitHubReleaseV0/task.json (hunk 2 above)
Comment (romil07) [reply Tasks/DownloadGitHubReleaseV0/task.json]: What is this?

File: Tasks/DownloadGitHubReleaseV0/task.json (hunk 1 above)
Comment (romil07) [reply Tasks/DownloadGitHubReleaseV0/task.json]: We won't make it 149 as lova suggested. It will remain 148 most probably unless my check in is delayed.

File: Tasks/DownloadGitHubReleaseV0/task.json (hunk 2 above)
Comment (gykuma) [reply Tasks/DownloadGitHubReleaseV0/task.json]: Not sure where it will pick the **name** from, do we expect input in **version** to be passed as **searchText**?

Comment (gykuma) [reply Tasks/DownloadGitHubReleaseV0/task.json]: Not required as discussed.

File: Tasks/DownloadGitHubReleaseV0/task.loc.json (hunk 3 above)
Comment (omeshp) [inline Tasks/DownloadGitHubReleaseV0/task.loc.json]:
```suggestion
    "Minor": 149,
```

File: Tasks/DownloadGitHubReleaseV0/task.json [hunk 5]
```diff
@@ -94,6 +96,15 @@
         "repositoryName": "$(userRepository)",
         "releaseByTagName": "$(defaultVersionType)"
       }
+    },
+    {
+      "target": "userRepositorySearch",
+      "endpointId": "$(connection)",
+      "dataSourceName": "SearchRepositoriesByNameForTasks",
+      "resultTemplate": "{ \"Value\" : \"{{full_name}}\", \"DisplayValue\" : \"{{full_name}}\" }",
```
Comment (Lovakumar) [inline Tasks/DownloadGitHubReleaseV0/task.json]: "resultSelector": "jsonpath:$['items'][*]"

File: Tasks/DownloadGitHubReleaseV0/task.json [hunk 6]
```diff
@@ -94,6 +96,15 @@
         "repositoryName": "$(userRepository)",
         "releaseByTagName": "$(defaultVersionType)"
       }
+    },
+    {
+      "target": "userRepositorySearch",
+      "endpointId": "$(connection)",
+      "dataSourceName": "SearchRepositoriesByNameForTasks",
```
Comment (Lovakumar) [inline Tasks/DownloadGitHubReleaseV0/task.json]: Change this to SearchRepositoriesByName

File: Tasks/DownloadBuildArtifactsV0/task.json [hunk 7]
```diff
@@ -198,7 +198,7 @@
             "resultSelector": "jsonpath:$.value[?(@.quality=='definition')]",
             "parameters": {
                 "project": "$(project)",
-                "name": "$(name)"
+                "name": "$(searchText)"
```
Comment (Lovakumar) [inline Tasks/DownloadBuildArtifactsV0/task.json]: This may be an issue with TFS AT DT mismatch case. Ensure that the UI is still sending both name and searchText.

File: Tasks/GitHubReleaseV0/task.json [hunk 8]
```diff
@@ -181,14 +182,23 @@
...
-    "sourceDefinitions": [
+    "dataSourceBindings": [
         {
             "target": "repositoryName",
-            "endpoint": "/$(system.teamProject)/_apis/sourceProviders/GitHub/repositories?serviceEndpointId=$(gitHubConnection)",
-            "selector": "jsonpath:$.repositories[*].id",
-            "keySelector": "jsonpath:$.repositories[*].id",
-            "authKey": "tfs:teamfoundation"
-        }
+            "endpointId": "$(gitHubConnection)",
```
Comment (Lovakumar) [inline Tasks/GitHubReleaseV0/task.json]: Get this reviewed with Akbar

File: Tasks/GitHubReleaseV0/task.json [hunk 9]
```diff
@@ -15,7 +15,7 @@
     "version": {
         "Major": 0,
         "Minor": 0,
-        "Patch": 4
+        "Patch": 5
```
Comment (mdmdakbar) [inline Tasks/GitHubReleaseV0/task.json]: Is this change minor or Patch? Also, the task version is already 0.0.5, please do the needful here

Comment (romil07) [reply Tasks/GitHubReleaseV0/task.json]: I just bumped up the version which is being maintained. I don't know if you want to upgrade the minor one or not. Your call.

File: Tasks/DownloadGitHubReleaseV0/task.json (hunk 1 above)
Comment (Lovakumar) [reply Tasks/DownloadGitHubReleaseV0/task.json]: This should remain as M148.

File: Tasks/DownloadGitHubReleaseV0/task.loc.json [hunk 10]
```diff
@@ -10,7 +10,7 @@
   "demands": [],
   "version": {
     "Major": 0,
-    "Minor": 148,
+    "Minor": 149,
```
Comment (Lovakumar) [inline Tasks/DownloadGitHubReleaseV0/task.loc.json]: Should remain as M148

File: Tasks/GitHubReleaseV0/task.json [hunk 11]
```diff
@@ -15,7 +15,7 @@
     "version": {
         "Major": 0,
         "Minor": 0,
-        "Patch": 4
+        "Patch": 6
```
Comment (Lovakumar) [inline Tasks/GitHubReleaseV0/task.json]: This looks to be wrong versioning to me. Update Akbar about it.

Comment (mdmdakbar) [reply Tasks/GitHubReleaseV0/task.json]: Thanks @romil07 for making this change at priority
